task/options/options.go

```diff
@@ -40,22 +45,129 @@ type Options struct {
 
 	// Every represents a fixed period to repeat execution.
 	// this can be unmarshaled from json as a string i.e.: "1d" will unmarshal as 1 day
-	Every time.Duration `json:"every,omitempty"`
+	Every Duration `json:"every,omitempty"`
 
 	// Offset represents a delay before execution.
 	// this can be unmarshaled from json as a string i.e.: "1d" will unmarshal as 1 day
-	Offset *time.Duration `json:"offset,omitempty"`
+	Offset *Duration `json:"offset,omitempty"`
 
 	Concurrency *int64 `json:"concurrency,omitempty"`
 
 	Retry *int64 `json:"retry,omitempty"`
 }
 
+type Duration struct {
+	Node ast.DurationLiteral
+}
+
+func (a Duration) String() string {
+	return ast.Format(&a.Node)
+}
+
+func (a *Duration) Parse(s string) error {
+	q, err := parseSignedDuration(s)
+	if err != nil {
+		return err
+	}
+	a.Node = *q
+	return nil
+}
+
+func MustParseDuration(s string) (dur *Duration) {
+	dur = &Duration{}
+	if err := dur.Parse(s); err != nil {
+		panic(err)
+	}
+	return dur
+}
+
+// parseSignedDuration will convert a string into a possibly negative DurationLiteral.
+// TODO(docmerlin): switch to the one in flux
+func parseSignedDuration(lit string) (*ast.DurationLiteral, error) {
+	r, s := utf8.DecodeRuneInString(lit)
+	if r == '-' {
+		d, err := parser.ParseDuration(lit[s:])
+		if err != nil {
+			return nil, err
+		}
+		for i := range d.Values {
+			d.Values[i].Magnitude = -d.Values[i].Magnitude
+		}
+		d.BaseNode = ast.BaseNode{}
+		return d, nil
+	}
+	d, err := parser.ParseDuration(lit)
+	if err != nil {
+		return nil, err
+	}
+	d.BaseNode = ast.BaseNode{}
+	return d, err
+}
+
+func (a *Duration) UnmarshalText(text []byte) error {
+	q, err := parseSignedDuration(string(text))
+	if err != nil {
+		return err
+	}
+	a.Node = *q
+	return nil
+}
+
+func (a Duration) MarshalText() ([]byte, error) {
+	return []byte(a.String()), nil
+}
+
+func (a Duration) MartialJSON() ([]byte, error) {
+	q := a.String()
+	res := make([]byte, len(q)+2)
+	res[0] = '"'
+	res[len(res)-1] = '"'
+	copy(res[1:len(res)-1], q)
+	return res, nil
+}
+
+// IsZero checks if each segment of the duration is zero, it doesn't check if the Duration sums to zero, just if each internal duration is zero.
+func (a *Duration) IsZero() bool {
+	for i := range a.Node.Values {
+		if a.Node.Values[i].Magnitude != 0 {
+			return false
+		}
+	}
+	return true
+}
+
+func (a *Duration) UnmartialJSON(b []byte) error {
+	if len(b) < 2 {
+		return errors.New("duration json must be at least 2 bytes long including the quotes")
+	}
+	x, err := parseSignedDuration(string(b[1 : len(b)-1]))
```

Comment:
I would probably rely on json.Unmarshal here too, like 

```go
var s string
if err := json.Unmarshal(b, &s); err != nil {
  return err
}

x, err := parseSignedDuration(s)
// ...
```

If we even need it, given the presence of MarshalText.